test_rules_scala.sh

```diff
@@ -16,13 +16,12 @@ $runner bazel build test/...
 #$runner bazel build "test/... --all_incompatible_changes"
 $runner bazel test test/...
 $runner bazel test third_party/...
-# UnusedDependencyChecker doesn't work with strict_java_deps
```

Comment:
@ittaiz not sure why we needed `-test:UnusedDependencyChecker` but as the comment that unused/strict doesn't go together isn't relevant anymore, so removed them in this block of code. Though it seems even in the past it would still compile and just not run unused deps checker (and there weren't unused deps in that rule as far as I could tell).